module/VuFind/src/VuFind/Auth/Shibboleth.php

```diff
@@ -229,17 +305,8 @@ public function getSessionInitiator($target)
     public function isExpired()
     {
         $config = $this->getConfig();
-        if (isset($config->Shibboleth->username)
-            && isset($config->Shibboleth->logout)
-        ) {
-            // It would be more proper to call getServer on a Laminas request
-            // object... except that the request object doesn't exist yet when
-            // this routine gets called.
-            $username = isset($_SERVER[$config->Shibboleth->username])
-                ? $_SERVER[$config->Shibboleth->username] : null;
-            return empty($username);
-        }
-        return false;
+        $sessionId = $this->getAttribute($this->request, $this->shibSessionId);
```

Comment:
I still don't think this is correct. The fact that we have a Shibboleth logout URL does not mean that the VuFind session should expire when the Shibboleth session is gone. If you want the Shibboleth session expiration to expire the VuFind session too, I think there should be a separate setting (that defaults to false) to indicate that.

This is somewhat related to the fact that verifying the Shibboleth session for each request is expensive. For a busy server the Shibboleth daemon can become a bottleneck, and I wouldn't recommend using `Require shibboleth` globally (see https://vufind.org/wiki/configuration:shibboleth for more information). This means that the Shibboleth session information is not always available anyway.